homeassistant/components/homekit_controller/config_flow.py

```diff
@@ -307,14 +277,51 @@ async def async_step_pair(self, pair_info=None):
                 self.finish_pairing = None
                 errors["pairing_code"] = "pairing_failed"
 
-        if errors and "base" in errors:
-            return self.async_show_form(step_id="try_pair_later", errors=errors)
+        if not self.finish_pairing:
+            # Its possible that the first try may have been busy so
+            # we always check to see if self.finish_paring has been
+            # set.
+            discovery = await self.controller.find_ip_by_device_id(self.hkid)
+
+            try:
+                self.finish_pairing = await discovery.start_pairing(self.hkid)
+                self.try_later_reason = None
+
+            except aiohomekit.BusyError:
+                # Already performing a pair setup operation with a different
+                # controller
+                self.try_later_reason = "busy_error"
+                return await self.async_step_try_pair_later()
```

Comment:
Seems we only use a different step to get a different description. Can we set description placeholders instead to generate different descriptions for the pair step?

Reply:
If we don't want to allow the user to input something in the form this time, adjust the data schema of the form. We don't need to go to a different step.

Reply:
placeholders can't be translated

Reply:
Ok, then I think we need three steps, one for each `try_later_reason` to avoid using errors for this.

Reply:
Adjusted to be three steps